src/transaction/signer.cpp

```diff
@@ -1,19 +1,24 @@
-#include <string.h>
+#include <cstring>
 #include <sodium.h>
+#include <stdexcept>
 
+#include "params.h"
 #include "strchr.h"
 #include "transaction/signer.h"
 
 
-Signer::Signer(bytes secretKey) :
+Signer:: Signer(bytes const &secretKey) :
         m_sk(secretKey)
-{}
+{
+    if (secretKey.size() != crypto_sign_SECRETKEYBYTES)
+        throw std::length_error(ERROR_MSG_KEY_BYTES_SIZE);
+}
 
 std::string Signer::getSignature(std::string const &message) const
 {
     //TODO: Put this code in a wrapper for libsodium library
     unsigned long long const msgLength = message.length();
-    unsigned char *msg = new unsigned char[msgLength + 1ULL];
+    auto *msg = new unsigned char[msgLength + 1ULL];
```

Comment:
Perhaps add a comment explaining why the need to create a new buffer (`msg`) and copy into it? Or is this just for casting purposes? If so, can it be done with a cast?

Reply:
Yes, the need for the new buffer is for casting purpose only -> since libsodium works with `unsigned char*` instead of `byte array`.
I'll use [this](https://github.com/ElrondNetwork/elrond-sdk-erdcpp/pull/4#discussion_r646188004) suggestion and change the current implementation with a cast in [this](https://elrondnetwork.atlassian.net/browse/EN-10130) PR